src/main/java/entities/Post.java

```diff
@@ -12,15 +13,34 @@
     private User owner;
     private Recipe recipe;
     private String category;
+    private int idPost;
     private LocalDateTime postedTime;
 
-    public Post(User owner, LocalDateTime postedTime, Recipe recipe, String category) {
+    public Post(User owner, LocalDateTime postedTime, Recipe recipe, String category, int pId) {
         this.likedUsers = new ArrayList<>();
         this.comments = new ArrayList<>();
         this.owner = owner;
         this.recipe = recipe;
         this.category = category;
         this.postedTime = postedTime;
+        this.idPost = pId;
+    }
+
+    public int getIdPost() {
+        return this.idPost;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Post post = (Post) o;
+        return idPost == post.idPost;
```

Comment:
If you agree that we would only be comparing Posts, do you think we could reduce this function to one line? (i.e. return postID == post.postID